src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -68,7 +68,7 @@ public sealed partial class JsonDocument
         //   * 31 bits for token offset
         // * Second int
         //   * Top bit is unassigned / always clear
-        //   * 31 bits for the token length (always 1, effectively unassigned)
```

Comment:
I think it might make sense to validate if this comment is consistent with the current implementation, just to make sure we're not accidentally regressing something.

Reply:
FYI: For me it seemed that the comment was wrong because previously it wasn't always 1 for `StartObject`. Instead it was set to the same value as the third int (number of rows until the next value). See first and last line here:

https://github.com/dotnet/runtime/blob/fc7a8e62c4f0351477fd7423e67e7a541e9c7ca9/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L990-L994

Reply:
Looks like the comment was always wrong (at least, was wrong in the initial merge, maybe it was right when I wrote it during feature development).

Nothing noticed or cared that the value was wrong, because nothing read the value until DeepEquals was added... and I guess it didn't notice that it was wrong, because it was the "same wrong" when two things were equal.

Reply:
Which makes think that we might want to backport this change to 9. Even though functional correctness of `DeepEquals` is probably not impacted, it does mean that it will always fall back to the more expensive property comparison.

Reply:
It is possible to have equal number of rows when the number of properties are not equal:

- **Example 1**

  10 rows and 2 properties
  ```json
  { "a": [1,2,3,4], "b": 4 }
  ```
  10 rows but 3 properties
  ```json
  { "a": [1,2], "b": 3, "c": 4 }
  ```

- **Example 2** 

  5 rows and 1 property:
  ```json
  { "a": [1] }
  ```

  5 rows and 2 properties
  ```json
  { "a": 1, "b": 2 }
  ```

But I would say it is very unlikely that two unequal values accidentally have the same number of rows.

Reply:
Thanks, I think we should try to backport this.